addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitChangeListener.java

```diff
@@ -79,8 +137,24 @@ public synchronized void setBridge(HomekitRoot bridge) {
 
     public synchronized void setItemRegistry(ItemRegistry itemRegistry) {
         this.itemRegistry = itemRegistry;
-        itemRegistry.addRegistryChangeListener(this);
-        itemRegistry.getAll().forEach(item -> added(item));
+        maybeInitialize();
+    }
+
+    /**
+     * Call after itemRegistry and settings are specified to initialize homekit devices
+     */
+    private void maybeInitialize() {
```

Comment:
Is there a reason for this class not being a `@Component`?
The OSGi framework would resolve your requirements (ItemRegistry + HomekitSettings)
and you can use the `activated` method to do do what you currently do in `maybeInitialize`.

`HomekitSettings` would be a component in that scenario as well, which it probably isn't I guess?